include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -783,27 +800,43 @@ __parallel_radix_sort(_ExecutionPolicy&& __exec, _Range&& __in_rng)
     auto __out_rng = oneapi::dpl::__ranges::all_view<_T, __par_backend_hetero::access_mode::read_write>(
         __out_buffer_holder.get_buffer());
 
-    // iterations per each bucket
-    assert("Number of iterations must be even" && __radix_iters % 2 == 0);
-    // TODO: radix for bool can be made using 1 iteration (x2 speedup against current implementation)
-    sycl::event __iteration_event{};
-    for (::std::uint32_t __radix_iter = 0; __radix_iter < __radix_iters; ++__radix_iter)
+    sycl::event __event{};
+
+    if (__n < 512)                                                      // v--- block size
+        __event = __group_radix_sort<__i_kernel_name<_RadixSortKernel, 0>, 1, __radix_bits, __is_ascending>(
+            __exec.queue(), __in_rng, __out_rng, __wg_size);
+    else if (__n == 512)
+        __event = __group_radix_sort<__i_kernel_name<_RadixSortKernel, 1>, 2, __radix_bits, __is_ascending>(
+            __exec.queue(), __in_rng, __out_rng, __wg_size);
```

Comment:
Should `512` be based on `__wg_size` in some fashion?  or is 512 the correct number here for all devices?

Reply:
Due to the code was changed, the comment is not actual any more.